closure/templates/closure_js_template_library.bzl

```diff
@@ -47,6 +51,19 @@ def _impl(ctx):
         for f in dep.closure_js_library.descriptors.to_list():
             args += ["--protoFileDescriptors=%s" % f.path]
             inputs.append(f)
+
+    plugin_deps = depset(
```

Comment:
Looking at this, making `SoyPluginInfo.generator.deps` a depset of `JavaInfo` was a bad idea (of me) :/. Would you mind changing it to `runtime = java_common.merge([...])` [1]? Then, this can be simplified to
```python
plugin_transitive_deps = depset(
    transitive = [m[SoyPluginInfo].generator.runtime.transitive_runtime_deps for m in ctx.attr.plugin_modules],
).to_list()
```

Also, generally, you want to avoid flattening depsets into lists as much as possible. The nice thing about depsets is that constructing them is very cheap (Should be `O( len(direct) + len(transitive) )`), but to create a list from them, Bazel has to perform expensive duplicate checks which can really slow-down builds.

[1] https://docs.bazel.build/versions/master/skylark/lib/java_common.html#merge

Reply:
That seems much better, good idea. I didn't know about `java_common.merge`